cloud-aws/src/main/resources/templates/aws-cf-stack-freeipa.ftl

```diff
@@ -0,0 +1,236 @@
+      "Type" : "String",
+      "MinLength": "15",
+      "MaxLength": "255",
+      "AllowedPattern" : "subnet-[a-z0-9]*(?:,subnet-[a-z0-9]*)*"
+    },
+
+    "StackOwner" : {
+      "Description" : "The instances will have this parameter as an Owner tag.",
+      "Type" : "String",
+      "MinLength": "1",
+      "MaxLength": "50"
+    },
+
+    "stackowner" : {
+       "Description" : "The instances will have this parameter as an owner tag.",
+       "Type" : "String",
+       "MinLength": "1",
+       "MaxLength": "200"
+    },
+
+    "CBGateWayUserData" : {
+      "Description" : "Gateway user data to be executed",
+      "Type" : "String",
+      "MinLength": "9",
+      "MaxLength": "4096"
+    },
+
+    "CBGateWayUserData1" : {
+      "Description" : "Gateway user data to be executed (continued)",
+      "Type" : "String",
+      "MinLength": "0",
+      "MaxLength": "4096"
+    },
+
+    "CBGateWayUserData2" : {
+      "Description" : "Gateway user data to be executed (continued)",
+      "Type" : "String",
+      "MinLength": "0",
+      "MaxLength": "4096"
+    },
+
+    "CBGateWayUserData3" : {
+      "Description" : "Gateway user data to be executed (continued)",
+      "Type" : "String",
+      "MinLength": "0",
+      "MaxLength": "4096"
+    },
+
+    "KeyName": {
+      "Description" : "Name of an existing EC2 KeyPair to enable SSH access to the instances",
+      "Type": "String",
+      "MinLength": "1",
+      "MaxLength": "255",
+      "AllowedPattern" : "[\\x20-\\x7E]*",
+      "ConstraintDescription" : "can contain only ASCII characters."
+    },
+
+    "AMI" : {
+      "Description" : "AMI that's used to start instances",
+      "Type" : "String",
+      "MinLength": "12",
+      "AllowedPattern" : "ami-[a-z0-9]*",
+      "ConstraintDescription" : "must follow pattern: ami-xxxxxxxx"
+    },
+
+    "RootDeviceName" : {
+      "Description" : "Name of the root device that comes with the AMI",
+      "Type" : "String",
+      "MinLength": "8",
+      "MaxLength": "12"
+    }
+
+  },
+
+  "Resources" : {
+
+    <#if mapPublicIpOnLaunch>
+        <#list gatewayGroups as group>
+            <#list 1..group.instanceCount as nth>
+                "EIP${group.groupName?replace('_', '')}${nth}" : {
+                   "Type" : "AWS::EC2::EIP",
+                   "Properties" : {
+                      "Domain" : "vpc"
+                   }
+                },
+            </#list>
+        </#list>
+    </#if>
+
+    <#list instanceGroups as group>
+    "${group.autoScalingGroupName}" : {
+      "Type" : "AWS::AutoScaling::AutoScalingGroup",
+      "Properties" : {
+        <#if group.subnetId??>
+        "VPCZoneIdentifier" : [ "${group.subnetId}" ],
+        <#else>
+        "VPCZoneIdentifier" : [{ "Ref" : "SubnetId" }],
+        </#if>
+        "LaunchConfigurationName" : { "Ref" : "AmbariNodeLaunchConfig${group.groupName?replace('_', '')}" },
```

Comment:
Could you rename `AmbariNodeLaunchConfig`? Or is there a reason to keep it?

Reply:
unfortunately if we change this here then we should take care with the backward compatibility